server/chat-plugins/youtube.ts

```diff
@@ -0,0 +1,422 @@
+/**
+ * Youtube room chat-plugin.
+ * Supports adding channels and selecting a random channel.
+ * Also supports showing video data on request.
+ * Written by mia-pi, with some code / design concepts from Asheviere.
+ */
+
+import * as https from 'https';
+import {FS} from '../../lib/fs';
+const ROOT = 'https://www.googleapis.com/youtube/v3/';
+const CHANNEL = `${ROOT}channels`;
+const STORAGE_PATH = 'config/chat-plugins/youtube.json';
+
+let channelData: AnyObject;
+
+try {
+	channelData = JSON.parse(FS(STORAGE_PATH).readSync()) || {};
+} catch (e) {
+	channelData = {};
+}
+
+export const YouTube = new class {
+	interval: NodeJS.Timeout | null;
```

Comment:
wouldnt this be better suited as `NodeJS.Timer`?

Reply:
I don't think so?
VSCode says setInterval is `NodeJS.Timeout`.

Reply:
It doesn't matter. Checking `@types/node`'s source code, `Timeout` is a concrete class that implements `Timer` interface.

Reply:
Alright.